mava/networks.py

```diff
@@ -159,6 +163,52 @@ def __call__(self, obs_embedding: chex.Array, observation: Observation) -> tfd.I
         )
 
 
+class RecQNetwork(nn.Module):
+    """Recurrent Q-Network."""
+
+    pre_torso: nn.Module
+    post_torso: nn.Module
+    num_actions: int
+    hidden_state_dim: int = 128
+
+    @nn.compact
+    def get_q_values(
+        self,
+        hidden_state: chex.Array,
+        observations_resets: RNNObservation,
+    ) -> chex.Array:
+        """Forward pass to obtain q values."""
+
+        obs, resets = observations_resets
+
+        embedding = self.pre_torso(obs.agents_view)
```

Comment:
Adding in check for centralized critic
```suggestion
        if self.centralised_critic:
            if not isinstance(observation, ObservationGlobalState):
                raise ValueError("Global state must be provided to the centralised critic.")
            # Get global state in the case of a centralised critic.
            observation = observation.global_state
        else:
            # Get single agent view in the case of a decentralised critic.
            observation = observation.agents_view

        embedding = self.pre_torso(obs.agents_view)
```

Reply:
Where does a centralised critic appear in Q-learning? I am unfamiliar. Only Qmix ever uses global state.

Reply:
Do we want to switch between the environment being partially observable and fully observable?

Reply:
Offline: leave this for when QMix is being merged.